### PR TITLE
qml: add transaction inputs in TxDetails and …

### PR DESCRIPTION
**Pull request by accumulator**
ConfirmTxDialog, RbfBumpFeeDialog, RbfCancelDialog, CpfpBumpFeeDialog and allow collapse/expand of input/output lists by clicking on label

### CODE REVIEW

**Comment by SomberNight** [comment]
Looks nice.

Could we apply the same green/yellow colouring to the input addresses that the outputs have?
That is, if the input address is known, and it is a change address, make it yellow, and if it is non-change ismine, make it green.

**Comment by accumulator** [comment]
> Could we apply the same green/yellow colouring to the input addresses that the outputs have? That is, if the input address is known, and it is a change address, make it yellow, and if it is non-change ismine, make it green.

Yeah, I had that initially, but it looked a bit off.

**Comment by accumulator** [comment]
added color to inputs again.
with fontsize the same for inputs and outputs it looks better.

File: electrum/gui/qml/qetxfinalizer.py [hunk 1]
```diff
@@ -210,14 +222,38 @@ def update_from_tx(self, tx):
         self.fee = QEAmount(amount_sat=int(fee))
         self.feeRate = f'{feerate:.1f}'
 
+        self.update_inputs_from_tx(tx)
         self.update_outputs_from_tx(tx)
```
Comment (SomberNight) [inline electrum/gui/qml/qetxfinalizer.py:226]: You need to do something similar in `QETxCpfpFeeBumper.update()`, near (new) line 868.
There only `update_outputs_from_tx` is called atm, so either call either both (`update_inputs_from_tx` and `update_outputs_from_tx`),
or maybe call `update_from_tx` instead, though the other side-effects of that are not relevant.

**Review by SomberNight — COMMENTED** [review]
Skimmed the code, and then played around it with a bit.

I think it's much better than status quo. Even forgetting about the inputs, I really like seeing the output indices.

Also, and I guess this might have been your motivation, for advanced users it's useful to see the number of inputs when creating/bumping txs.

File: electrum/gui/qml/qetxfinalizer.py (hunk 1 above)
Comment (accumulator) [reply electrum/gui/qml/qetxfinalizer.py:226]: thx, good catch

**Comment by Jircs1** [comment]
https://github.com/spesmilo/electrum/pull/8772#issue-2054098522

**Review by Jircs1 — COMMENTED** [review]
accumulator:qml_tx_inputs

**Pull request merged**